scripts/validate-requirements.sh

```diff
@@ -0,0 +1,274 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage:
+# bash scripts/validate-requirements.sh "END_USER_EMAIL" "ORGANIZATION_ID" "BILLING_ACCOUNT_ID"
+
+# -------------------------- Variables --------------------------
+# Expected versions of the installers
+TF_VERSION="0.13.7"
+GCLOUD_SDK_VERSION="391.0.0"
+GIT_VERSION="2.25.1"
+
+# Input variables
+END_USER_CREDENTIAL=""
+ORGANIZATION_ID=""
+BILLING_ACCOUNT=""
+
+# Collect the errors
+ERRORS=""
+# -------------------------- Funcions ---------------------------
+
+# Compare two semantic versions
+function compare_version(){
+    if [[ "$1" == "$2" ]]; then
+        echo 0
+        return 0
+    fi
+
+    local IFS=.
+    local i
+    local version1=("$1")
+    local version2=("$2")
+    for ((i=${#version1[@]}; i<${#version2[@]}; i++))
+    do
+        version1[i]=0
+    done
+    for ((i=0; i<${#version1[@]}; i++))
+    do
+        if [[ -z ${version2[i]} ]]
+        then
+            version2[i]=0
+        fi
+        if [[ ${version1[i]} > ${version2[i]} ]]
+        then
+            echo 1
+            return 1
+        fi
+        if [[ ${version1[i]} < ${version2[i]} ]]
+        then
+            echo 2
+            return 2
+        fi
+    done
+    return 0
+}
+
+# Validate the Terraform installation and version
+function validate_terraform(){
+
+    if [ ! "$(command -v terraform )" ]; then
+        echo "Terraform not found."
+        echo "Visit https://learn.hashicorp.com/tutorials/terraform/install-cli and follow the instructions to install Terraform."
+        ERRORS+=$'Terraform not found\n'
+    else
+        TERRAFORM_CURRENT_VERSION=$(terraform version -json | jq -r .terraform_version)
+        if [ "$(compare_version "$TERRAFORM_CURRENT_VERSION" "$TF_VERSION")" -ne 0 ]; then
+            echo "An incompatibly was found in Terraform version."
+            echo "Terraform version $TF_VERSION is required."
+            echo "Visit https://learn.hashicorp.com/tutorials/terraform/install-cli and follow the instructions to install Terraform."
+            ERRORS+=$'Terraform version is incompatible.\n'
+        fi
+    fi
+}
+
+# Validate the Google Cloud SDK installation and version
+function validate_gcloud(){
+    if [ ! "$(command -v gcloud)" ]; then
+        echo "gcloud CLI not found."
+        echo "Visit https://cloud.google.com/sdk/docs/install and follow the instructions to install gcloud CLI."
+        ERRORS+=$'gcloud not found.\n'
+    else
+        GCLOUD_CURRENT_VERSION=$(gcloud version --format=json | jq -r '."Google Cloud SDK"')
+        if [ "$(compare_version "$GCLOUD_CURRENT_VERSION" "$GCLOUD_SDK_VERSION")" -eq 2 ]; then
+            echo "An incompatibly was found in gcloud version."
```

Comment:
nit:
```suggestion
            echo "An incompatible gcloud version was found."
```

Reply:
A few places where this can be fixed below. I might even move this to a func that takes in install URL and minimum version and echoes this warning.